lib/pbench/agent/utils.py

```diff
@@ -210,6 +210,44 @@ class RedisServerCommon(BaseServer):
     local_host = None
 
 
+def check_failed_tool_init(tms: dict) -> str:
+    """checks if there are any tool failures in the Tool-Meisters
+
+    Args:
+        tms: Tool Meister data and metadata published by redis.
+
+    returns "success" when there are no failed_tools else "failure".
+    """
+    for _, tm in tms.items():
+        if len(tm["failed_tools"]) > 0:
+            return "failed"
+    return "success"
+
+
+def log_failed_tool_init(logger: logging.Logger, tms: dict):
+    """logs the tool failures in the Tool-Meisters
+
+    Args:
+        tms: Tool Meister data and metadata extracted from the redis.
+    """
+    failure_count = 0
+    for entry in tms.values():
+        if len(entry["failed_tools"]) == 0:
+            continue
```

Comment:
Per [PEP-8](https://peps.python.org/pep-0008/#programming-recommendations), this would be better expressed as
```
        if not entry["failed_tools"]:
            continue
```